Chuck-1102-word-game_start.py

```diff
@@ -0,0 +1,179 @@
+# On this project I have collaborated with the following student: Poon (Class 1102)
+
+def main():
+    secret_word = choose_word(word_list)
+    print(game_loop(secret_word))
+
+# -----------------------------------
+# Helper code
+# You don't need to understand this helper code,
+# but you will have to know how to use the functions
+# (so be sure to read the docstrings!)
+
+import random
+import string
+
+WORDLIST_FILENAME = "word_list.txt"
+
+def load_words():
+    """
+    Returns a list of valid words. Words are strings of lowercase letters.
+    
+    Depending on the size of the word list, this function may
+    take a while to finish.
+    """
+    print("Reading word_list file...")
+    # inFile: file
+    inFile = open(WORDLIST_FILENAME, 'r')
+    # line: string
+    line = inFile.readline()
+    # word_list: list of strings
+    word_list = line.split()
+    print(len(word_list), "words found")
+    return word_list
+
+def choose_word(word_list):
+    """
+    word_list (list): list of words (strings)
+    Returns a word from word_list at random
+    """
+    return random.choice(word_list)
+
+# end of helper code
+# -----------------------------------
+
+# Load the list of words into the variable word_list
+# so that it can be accessed from anywhere in the program
+word_list = load_words()
+
+def is_word_guessed(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list, what letters have been guessed so far
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    # FILL IN YOUR CODE HERE...
+    found_counter = 0
+    for letter in secret_word:
+        if letter in letters_guessed:
+            found_counter += 1
+    if found_counter == len(secret_word):
+        return True
+    else:
+        return False
+    pass
+
+### Testcases
+# print(is_word_guessed('house', ['a', 'e', 'u', 'c', 'r', 'h', 'o', 's']))   Returns: True
+# print(is_word_guessed('apple', ['a', 'e', 'i', 'k', 'p', 'r', 's']))   Returns: False
+# print(is_word_guessed('durian', ['h', 'a', 'c', 'd', 'i', 'm', 'n', 'r', 't', 'u']))   Returns: True
+# print(is_word_guessed('pineapple', []))   Returns: False 
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list, what letters have been guessed so far
+    returns: string, comprised of letters and underscores that represents
+      what letters in secret_word have been guessed so far.
+    '''
+    # FILL IN YOUR CODE HERE...
+    output_string = ""
+    for letter in secret_word:
+        if letter in letters_guessed:
+            output_string += letter
+        else:
+            output_string += "_"
```

Comment:
When inserting underscores into your string, it's a good idea to add at least a space after each 
one, so it's clear to the user how many unguessed letters are left in the string (compare the 
readability of 
____
with 
_ _ _ _ 
).